File: solutions/elixir/rna-transcription/1/lib/rna_transcription.ex
defmodule RnaTranscription do
  @doc """
  Guard for empty list
  """
  def to_rna([]) do
    []
  end

  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RnaTranscription.to_rna('ACTG')
  'UGAC'
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    IO.puts(dna)
    [head|tail] = dna
    rev_head =
      case head do
        71 -> 
          67
        67 ->
          71
        84 ->
          65
        65 ->
          85
      end
    [rev_head|RnaTranscription.to_rna(tail)]
  end
end

File: solutions/erlang/rna-transcription/1/src/rna_transcription.erl
-module(rna_transcription).

-export([to_rna/1]).


to_rna("") -> [];
to_rna(Strand) -> to_rna(Strand,"").

to_rna([], Acc) -> Acc;
to_rna([$A|Tail], Acc) -> to_rna(Tail, Acc ++ "U");
to_rna([$C|Tail], Acc) -> to_rna(Tail, Acc ++ "G");
to_rna([$G|Tail], Acc) -> to_rna(Tail, Acc ++ "C");
to_rna([$T|Tail], Acc) -> to_rna(Tail, Acc ++ "A");

to_rna([Head|Tail], Acc) -> erlang:error(badarg,[Head,Tail, Acc]).
